hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleToken.java

```diff
@@ -212,6 +216,7 @@ public void deserialize(SerializableDataInputStream in, int version) throws IOEx
 		treasury = in.readSerializable();
 		totalSupply = in.readLong();
 		decimals = in.readInt();
+		currentSerialNum = in.readLong();
```

Comment:
This has to be retrieved after the `maxSupply` retrieval. MerkleTokens of an older version do not have this field, which will resolve with an error.